spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/jdbc/AutoConfigureTestDatabase.java

```diff
@@ -60,6 +60,15 @@
 	 */
 	EmbeddedDatabaseConnection connection() default EmbeddedDatabaseConnection.NONE;
 
+	/**
+	 * The custom connection URL to use. If specified it will override the default URL
+	 * string in the embedded datasource. It is expected to be fully specified and no
+	 * string format replacements will be performed.
+	 * @return the custom connection url to use
+	 */
+	@PropertyMapping(skip = SkipPropertyMapping.ON_DEFAULT_VALUE)
+	String url() default "";
```

Comment:
I don't have any bright idea right now but this introduces a potential inconsistency combination where you could specify a connection type for HSQL and then an URL for H2. I wonder if there's a way to structure the config to avoid that.

Reply:
I am going to explore adding a `urlOptions` instead of a full url. I think that would satisfy the original "ask" as well as limit the possible inconsistency, as well as not lose the other features like auto names.